### PR TITLE
docs: add a page on hook configuration options

### PR DESCRIPTION
**Pull request by rokm**
Primarily intended as a registry of all options that are available in hooks provided by PyInstaller.

### CODE REVIEW

**Comment by rokm** [comment]
Required if we plan to add more options to hooks (e.g., #6024).

File: doc/hooks-config.rst [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+
+.. code-block:: python
+
+    a = Analysis(["program.py"],
+                ...,
+                hooksconfig={
+                    "hook1": {
+                        "option-list": ["entry1", "entry2"],
+                        "option-int": 42,
```
Comment (bwoodsend) [inline doc/hooks-config.rst:24]: The `option-list` and `option-int` kind of hints that there's a specific `option-[type]` format going on when there isn't - they key names are completely arbitrary. Maybe nonsense them up a bit?

Comment (bwoodsend) [reply doc/hooks-config.rst:24]: Or just `some-list-option` and `an-integer-option` would do.

Comment (rokm) [reply doc/hooks-config.rst]: Good idea; I've randomized those example names a bit.

File: doc/hooks-config.rst [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    binaries, datas, hiddenimports = ...
+
+    # Collect extra data
+    def hook(hook_api):
+        # Boolean option 'collect_extra_data'
+        if get_hook_config(hook_api, 'myhook', 'collect_extra_data'):
+            extra_datas = ...  # Collect extra data
+            hook_api.add_datas(extra_datas)
+
+
```
Comment (bwoodsend) [inline doc/hooks-config.rst:116]: Can we squeeze in an example setting the corresponding spec file option here too?

Comment (rokm) [reply doc/hooks-config.rst:116]: Done.

**Review by bwoodsend — APPROVED** [review]

**Pull request merged**